### PR TITLE
Fix `timeBucket` not taking effect in EqualsAndHashCode annotation.

### PR DESCRIPTION
**Pull request by zifeihan**
### Fix timeBucket not taking effect in EqualsAndHashCode annotation
- [ ] Add a unit test to verify that the fix works.
- [x] Explain briefly why the bug exists and how to fix it.

Fix `timeBucket` not taking effect in EqualsAndHashCode annotation, Because the hashcode and equals methods generated by EqualsAndHashCode do not contain the properties of the parent class, the data related to the topology map is inaccurate during aggregation. use https://github.com/zifeihan/friday look runtime class info in jvm.

before fix:
image
after fix:
image
image

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -49,6 +49,7 @@ Release Notes.
 * Fix the priority setting doesn't work of the ALS analyzers.
 * Fix bug that `endpoint-name-grouping.yml` is not customizable in Dockerized case.
 * Fix bug that istio version metric type on UI template mismatches the otel rule.
+* Fix `entityId` not taking effect in EqualsAndHashCode annotation
```
Comment (wu-sheng) [inline CHANGES.md:52]: I think it is `timestamp` is not working.

**Comment by wu-sheng** [comment]
Please add the UTs to verify the `equals`

**Comment by zifeihan** [comment]
> Please add the UTs to verify the `equals`

Ok, I will add test cases.

File: CHANGES.md (hunk 1 above)
Comment (zifeihan) [reply CHANGES.md]: done.

File: oap-server/server-core/src/test/java/org/apache/skywalking/oap/server/core/analysis/manual/relation/endpoint/EndpointCallRelation.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package org.apache.skywalking.oap.server.core.analysis.manual.relation.endpoint;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class EndpointCallRelation {
```
Comment (wu-sheng) [inline oap-server/server-core/src/test/java/org/apache/skywalking/oap/server/core/analysis/manual/relation/endpoint/EndpointCallRelation.java:6]:
```suggestion
public class EndpointCallRelationTest {
```

File: oap-server/server-core/src/test/java/org/apache/skywalking/oap/server/core/analysis/manual/relation/endpoint/EndpointCallRelationTest.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+package org.apache.skywalking.oap.server.core.analysis.manual.relation.endpoint;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class EndpointCallRelationTest {
+    @Test
+    public void testServiceInstanceRelationClientSideMetricsEquals() {
+        EndpointRelationServerSideMetrics thisObject = new EndpointRelationServerSideMetrics();
```
Comment (wu-sheng) [inline oap-server/server-core/src/test/java/org/apache/skywalking/oap/server/core/analysis/manual/relation/endpoint/EndpointCallRelationTest.java:27]: I think the codes are incorrect. Please fix.

Comment (zifeihan) [reply oap-server/server-core/src/test/java/org/apache/skywalking/oap/server/core/analysis/manual/relation/endpoint/EndpointCallRelationTest.java:27]: Thank you, fixed.

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**